src/Lib/package.cpp

```diff
@@ -31,3 +33,19 @@ const std::string Package::GetName(void) { return std::string("NetPanzer"); }
 const std::string Package::GetFullyQualifiedName(void) {
   return Package::GetName() + " " + Package::GetVersion();
 }
+
+#else
+
+#include "package.hpp"
+#include "test.hpp"
+
+int main(void)
+{
+  int len = Package::GetVersion().length();
+  assert(len >= 5 && len < 10);
+  assert(Package::GetName().compare("NetPanzer") == 0);
+
+  return 0;
+}
+
```

Comment:
are you saying we don't need to switch to Rust to have tests in the same file as the code!?! :)

Reply:
lol. Nope! I try to think of everything! But you're observant for sure, I did get the idea from rust. :)